profiles-sdk-examples/README.md

```diff
@@ -0,0 +1,191 @@
+```
+
+### JFrog Artifactory Setup
+
+1. Java 11 (use Openjdk, see the link in [Resources](#resources) section)
+1. JFrog Artifactory credentials (shared in LastPass with everyone in `Shared-Engineering` folder)
+1. Install IntelliJ IDEA with the latest Kotlin plugin enabled ([Intellij IDEA](https://www.jetbrains.com/idea/))
+1. Put JFrog Artifactory credentials in `$USER_HOME/.gradle/gradle.properties` file. See `gradle.properties.template` for instructions.
+
+### Developer Setup
+
+See [dev.md](./docs/dev.md) how to work with a local (developer) installation of the Cortex Profiles SDK.
+
+## Examples
+
+Examples are structured to build upon themselves and grow in complexity. Each provides their own instructions for
+running as well as additional context. The top level [main-app](./main-app/README.md) is a CLI wrapper around the other
+examples:
+
+* [Local Cortex Clients](./local-clients/README.md)
+* [Join Connections](./join-connections/README.md)
+* [DataSource Refresh](./datasource-refresh/README.md)
+* [Build Profiles](./build-profiles/README.md)
+
+[picocli](https://picocli.info/) is used by each example to create a minimal CLI application for running the example.
+Refer to the instructions in each example.
+
+## Skill Template
+
+<!-- TODO: Seems inefficient that we package the config in the docker image. We should be able to set some config
+    options in the payload, so we don't have to rebuild an image for a config update.
+-->
+
+The [Skill Template](./templates) directory contains files for packaging as [Cortex Job Skill](./templates/skill.yaml), where:
+* The input to the skill is a [JSON Payload](./templates/payload.json) with the path to [Spark Configuration File](https://spark.apache.org/docs/latest/submitting-applications.html) in the Skill's Docker container.
+* The output of the skill is the Job execution logs
+* The [docker image](./main-app/src/main/resources/Dockerfile) for the `main-app` uses
+  a [Spark Submit](https://spark.apache.org/docs/latest/submitting-applications.html) based wrapper to run the Spark
+  application. The resources for the spark-submit wrapper is in the [main-app/src/main/resources/python/](./main-app/src/main/resources/python)
+  directory and is necessary for packaging as a Skill.
+
+**Before creating the skill**, you will need to:
+* Update the [spark-conf.json](./main-app/src/main/resources/conf/spark-conf.json) with the CLI application command and other config options.
+* Verify skills `payload.json` file refers to the above Spark configuration file (in the built container).
+
+To build the skill, you will need to set the following environment variables:
+* `DOCKER_PREGISTRY_URL=xxxx` - Private Registry Url accessible from Cortex
+* `PROJECT_NAME=xxxx` - Name of the project to save the skill, action and types
+* `CORTEX_TOKEN=xxxx` - For connecting to Cortex
+
+Building the Skill
```

Comment:
I'm fine with numbering this. I kinda think this is just 4 steps 

1. Before Creating the Skill (Configure the Skill files and export Environment Variables)
2. Build the Skill Image
3. Save and Deploy the Skill
4. Invoke the Skill

(3) and (4) can be combined into a single Makefile command, so its a one step build. I think this would work fine with Bikash's ask to have a `make all`.

I don't know about the details of the VS Code Extension and Workspaces, but I'm guessing this isn't too far off? (I'm also wondering if we should recommend using VSCode for the sake of Extension, because we currently recommend and are using IntelliJ)